File: VB/Scaffolding.DetailCollections/DepartmentContextDataModel/DesignTime/DesignTimeEmployeeRepository.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Linq
Imports System.Data
Imports System.Data.Entity
Imports System.Linq.Expressions
Imports System.Collections.Generic
Imports Scaffolding.DetailCollections.Common.Utils
Imports Scaffolding.DetailCollections.Model
Imports Scaffolding.DetailCollections.Common.DataModel
Imports Scaffolding.DetailCollections.Common.DataModel.EntityFramework

Namespace Scaffolding.DetailCollections.DepartmentContextDataModel
	Public Class DesignTimeEmployeeRepository
		Inherits DesignTimeRepository(Of Employee, Integer)
		Implements IEmployeeRepository
		Public Sub New()
			MyBase.New(Function(x) x.EmployeeID)
		End Sub
	End Class
End Namespace

File: CS/Scaffolding.DetailCollections/DepartmentContextDataModel/DesignTime/DesignTimeEmployeeRepository.cs
using System;
using System.Linq;
using System.Data;
using System.Data.Entity;
using System.Linq.Expressions;
using System.Collections.Generic;
using Scaffolding.DetailCollections.Common.Utils;
using Scaffolding.DetailCollections.Model;
using Scaffolding.DetailCollections.Common.DataModel;
using Scaffolding.DetailCollections.Common.DataModel.EntityFramework;

namespace Scaffolding.DetailCollections.DepartmentContextDataModel {
    public class DesignTimeEmployeeRepository : DesignTimeRepository<Employee, int>, IEmployeeRepository {
        public DesignTimeEmployeeRepository()
            : base(x => x.EmployeeID) {
        }
    }
}